src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/AssessmentController.kt

```diff
@@ -144,31 +169,14 @@ class AssessmentController(
         }
 
         else -> {
-          val updated =
-            assessmentService.updateAssessment(
-              user,
-              assessmentId,
-              objectMapper.writeValueAsString(updateAssessment.data),
-            )
-          val assessmentValidationResult =
-            when (updated) {
-              is AuthorisableActionResult.Success -> updated.entity
-              is AuthorisableActionResult.NotFound -> throw NotFoundProblem(assessmentId, "Assessment")
```

Comment:
Looking at `AssessmentService.updateAssessment()`, it's not populating `assessmentId, "Assessment"` for `AuthorisableActionResult.NotFound()`, so you're loosing this context. I think this needs adding into `updateAssessment()`